src/ngraph_builder.cc

```diff
@@ -4187,14 +4187,63 @@ static Status TranslateSelectOp(
   TF_RETURN_IF_ERROR(
       GetInputNodes(ng_op_map, op, &ng_input1, &ng_input2, &ng_input3));
 
-  // broadcast to make all tensors same shape, as required by ngraph select op
-  std::tie(ng_input1, ng_input2) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input1, ng_input2));
-  std::tie(ng_input2, ng_input3) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input2, ng_input3));
+  if (ng_input2->get_shape() != ng_input2->get_shape()) {
+    return errors::InvalidArgument(
+        "Input tensors 2 and 3 should have same shape");
+  }
```

Comment:
lets add the other check: (rank of pred == 1) || (rank of pred > 1 && rank of pred == rank of input 1 && pred_shape[0] == input1[0])

We should never fail this check ever as TF will barf before we reach this stage, but its good to have as documentation. Lets have a test or a comment to this effect